### ISSUE TITLE
Generator not working

### ISSUE THREAD

**Issue by PatrickJGabriel** [yeoman/generator-bootstrap#42]
Hello, I am not entirely sure what the issue is, but after running 

`yo bootstrap`

and getting the subsequent menu to choose between CSS, SASS & LESS for my project, no matter which one I choose it doesn't seem to run. I'm in the directory in which I want yo to scaffold to, then run the command, yet nothing happens. Is there something I should be doing that i'm not?

**Comment by SBoudrias** [comment]
Nothing is outputted?

**Comment by PatrickJGabriel** [comment]
No, nothing at all. I have tried other generators to see if it was possibly my yo install in general that was the problem but my angular generator worked just fine.

**Comment by SBoudrias** [comment]
Okay, well maybe that generator is broken, we haven't touched the core of it in ~ 2 years.

Basically no one stepped up to be the maintainer of it.

**Comment by PatrickJGabriel** [comment]
Ah I see, on the yeoman website it says the generator was updated 6 months ago, but I see now here on Github that it was last update Nov of 2015. Well that's okay, thank you very much.

**Comment by SBoudrias** [comment]
Let's keep open. We/someone should fix it.

**Comment by NuroDev** [comment]
Agreed. Also getting this. Fresh install and generates no content at all.

**Comment by dawndiy** [comment]
run `npm install -g bower` first will fix this.

**Comment by osmarpetry** [comment]
@SBoudrias could I add the following line of code, in `./generator-bootstrap/app/index.js` to check if bower is installed?

```Generator.prototype.bootstrapFiles = function bootstrapFiles() {
   try {
    require.resolve('bower');
  } catch (e) {
    console.error('Bower is not found! Run "npm install -g bower"');
    process.exit(e.code);
  }
  // map format -> package name
  var packages = {
    css: 'bootstrap',
    sass: 'bootstrap-sass-official',
    less: 'components-bootstrap',
    stylus: 'bootstrap-stylus'
  };

  this.bowerInstall(packages[this.format], { save: true });
};
```
Or maybe just add the bower inside the `package.json`
```{
  "name": "generator-bootstrap",
  "version": "0.2.2",
  "description": "Yeoman generator for Bootstrap",
  "keywords": [
    "yeoman-generator",
    "framework",
    "component",
    "bootstrap",
    "ui",
    "css",
    "sass",
    "less"
  ],
  "author": "The Yeoman Team",
  "repository": "yeoman/generator-bootstrap",
  "scripts": {
    "test": "mocha"
  },
  "dependencies": {
    "bower": "^1.8.0",
    "yeoman-generator": "^0.18.9"
  },
  "devDependencies": {
    "mocha": "^2.3.3"
  },
  "license": "BSD-2-Clause"
}
```

I'm asking because I never did it before.

**Comment by SBoudrias** [comment]
@osmarpetry global modules are not `require()`able in Node. As such, this wouldn't fix the issue.

**Comment by osmarpetry** [comment]
@SBoudrias thank you

**Comment by emilianos** [comment]
I'm having the same issue, how did you solve this?

**Comment by osmarpetry** [comment]
@emilianos 
If you're using Mac or Linux type on terminal "sudo npm install -g bower"
If you're on Windows start de CMD as **administrator** and type "npm install -g bower"
